src/use-wake-lock.ts

```diff
@@ -68,6 +70,26 @@ export const useWakeLock = ({
     wakeLock.current && (await wakeLock.current.release());
   }, [isSupported]);
 
+  React.useEffect(() => {
+    if (reacquireOnPageVisible) {
+      const handleVisibilityChange = async () => {
+        if (wakeLock.current && document.visibilityState === 'visible') {
```

Comment:
@Deftera186 @jorisre hey folks, I might be missing something here, but shouldn't this check for `if (wakeLock.current === null && document.visibilityState === 'visible')` instead?

Navigating away from the page triggers `wakeLock.current.onrelease`, which then sets `wakeLock.current = null`.

Returning to the page then should check for `wakeLock.current === null`, before attempting to reacquire the lock.

Reply:
Good catch